Sources/Build/BuildDescription/ProductBuildDescription.swift

```diff
@@ -218,6 +219,7 @@ public final class ProductBuildDescription: SPMBuildCore.ProductBuildDescription
                     self.observabilityScope.emit(.swiftBackDeployError)
                 } else if self.buildParameters.targetTriple.isSupportingStaticStdlib {
                     args += ["-static-stdlib"]
+                    hasStaticStdlib = true
```

Comment:
Shouldn't shouldLinkStaticSwiftStdlib when !isSupportingStaticStdlib be an error?

Reply:
It seems a bit harder to adopt that approach here because the inclusion of `-static-stdlib` in the linker invocation also depends on the product type. If I understand your point correctly, I agree that there should be better validation around `shouldLinkStaticSwiftStdlib` (e.g. emitting an error if the product is a dynamic library) but imo that would be better suited to a follow-up PR since it's orthogonal to this change — what do you think?

Reply:
A follow-up PR is fine by me, but in that case would you mind adding a TODO or FIXME note in code comments? I hope this would be addressed sooner rather than later.